failing CI due to CFTimeIndex errors

the normal CI started to fail:
```
FAILED xarray/tests/test_cftimeindex.py::test_distant_cftime_datetime_sub_cftimeindex[365_day]
FAILED xarray/tests/test_cftimeindex.py::test_distant_cftime_datetime_sub_cftimeindex[360_day]
FAILED xarray/tests/test_cftimeindex.py::test_distant_cftime_datetime_sub_cftimeindex[julian]
FAILED xarray/tests/test_cftimeindex.py::test_distant_cftime_datetime_sub_cftimeindex[all_leap]
FAILED xarray/tests/test_cftimeindex.py::test_distant_cftime_datetime_sub_cftimeindex[366_day]
FAILED xarray/tests/test_cftimeindex.py::test_distant_cftime_datetime_sub_cftimeindex[gregorian]
FAILED xarray/tests/test_cftimeindex.py::test_distant_cftime_datetime_sub_cftimeindex[proleptic_gregorian]
```
<details><summary>traceback</summary>

```pytb
[gw2] linux -- Python 3.9.2 /usr/share/miniconda/envs/xarray-tests/bin/python

>   ???
E   TypeError: Expected unicode, got datetime.timedelta

pandas/_libs/tslibs/timedeltas.pyx:264: TypeError

During handling of the above exception, another exception occurred:

calendar = 'proleptic_gregorian'

    @requires_cftime
    @pytest.mark.parametrize("calendar", _CFTIME_CALENDARS)
    def test_distant_cftime_datetime_sub_cftimeindex(calendar):
        a = xr.cftime_range("2000", periods=5, calendar=calendar)
        with pytest.raises(ValueError, match="difference exceeds"):
>           a.date_type(1, 1, 1) - a

/home/runner/work/xarray/xarray/xarray/tests/test_cftimeindex.py:833: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/home/runner/work/xarray/xarray/xarray/coding/cftimeindex.py:581: in __rsub__
    return pd.TimedeltaIndex(other - np.array(self))
/usr/share/miniconda/envs/xarray-tests/lib/python3.9/site-packages/pandas/core/indexes/timedeltas.py:155: in __new__
    tdarr = TimedeltaArray._from_sequence_not_strict(
/usr/share/miniconda/envs/xarray-tests/lib/python3.9/site-packages/pandas/core/arrays/timedeltas.py:250: in _from_sequence_not_strict
    data, inferred_freq = sequence_to_td64ns(data, copy=copy, unit=unit)
/usr/share/miniconda/envs/xarray-tests/lib/python3.9/site-packages/pandas/core/arrays/timedeltas.py:957: in sequence_to_td64ns
    data = objects_to_td64ns(data, unit=unit, errors=errors)
/usr/share/miniconda/envs/xarray-tests/lib/python3.9/site-packages/pandas/core/arrays/timedeltas.py:1067: in objects_to_td64ns
    result = array_to_timedelta64(values, unit=unit, errors=errors)
pandas/_libs/tslibs/timedeltas.pyx:269: in pandas._libs.tslibs.timedeltas.array_to_timedelta64
    ???
pandas/_libs/tslibs/timedeltas.pyx:222: in pandas._libs.tslibs.timedeltas.convert_to_timedelta64
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   OverflowError: Python int too large to convert to C long

pandas/_libs/tslibs/timedeltas.pyx:167: OverflowError
```

</details>

This seems to coincide with the release of `pandas=1.2.4`, and this is indeed the only difference in the environment (I didn't try to reproduce, though).

cc @spencerkclark

--- a/xarray/coding/cftimeindex.py
+++ b/xarray/coding/cftimeindex.py
@@ -57,7 +57,12 @@ from .times import _STANDARD_CALENDARS, cftime_to_nptime, infer_calendar_name
 CFTIME_REPR_LENGTH = 19
 ITEMS_IN_REPR_MAX_ELSE_ELLIPSIS = 100
 REPR_ELLIPSIS_SHOW_ITEMS_FRONT_END = 10
-OUT_OF_BOUNDS_TIMEDELTA_ERRORS = (pd.errors.OutOfBoundsTimedelta, OverflowError)
+
+
+if LooseVersion(pd.__version__) > LooseVersion("1.1.0"):
+    OUT_OF_BOUNDS_TIMEDELTA_ERRORS = (pd.errors.OutOfBoundsTimedelta, OverflowError)
+else:
+    OUT_OF_BOUNDS_TIMEDELTA_ERRORS = (OverflowError, )
 
 
 def named(name, pattern):

--- a/xarray/coding/cftimeindex.py
+++ b/xarray/coding/cftimeindex.py
@@ -62,7 +62,7 @@ REPR_ELLIPSIS_SHOW_ITEMS_FRONT_END = 10
 if LooseVersion(pd.__version__) > LooseVersion("1.1.0"):
     OUT_OF_BOUNDS_TIMEDELTA_ERRORS = (pd.errors.OutOfBoundsTimedelta, OverflowError)
 else:
-    OUT_OF_BOUNDS_TIMEDELTA_ERRORS = (OverflowError, )
+    OUT_OF_BOUNDS_TIMEDELTA_ERRORS = (OverflowError,)
 
 
 def named(name, pattern):

--- a/xarray/coding/cftimeindex.py
+++ b/xarray/coding/cftimeindex.py
@@ -57,12 +57,7 @@ from .times import _STANDARD_CALENDARS, cftime_to_nptime, infer_calendar_name
 CFTIME_REPR_LENGTH = 19
 ITEMS_IN_REPR_MAX_ELSE_ELLIPSIS = 100
 REPR_ELLIPSIS_SHOW_ITEMS_FRONT_END = 10
-
-
-if LooseVersion(pd.__version__) > LooseVersion("1.2.3"):
-    OUT_OF_BOUNDS_TIMEDELTA_ERROR = pd.errors.OutOfBoundsTimedelta
-else:
-    OUT_OF_BOUNDS_TIMEDELTA_ERROR = OverflowError
+OUT_OF_BOUNDS_TIMEDELTA_ERRORS = (pd.errors.OutOfBoundsTimedelta, OverflowError)
 
 
 def named(name, pattern):
@@ -568,7 +563,7 @@ class CFTimeIndex(pd.Index):
         elif _contains_cftime_datetimes(np.array(other)):
             try:
                 return pd.TimedeltaIndex(np.array(self) - np.array(other))
-            except OUT_OF_BOUNDS_TIMEDELTA_ERROR:
+            except OUT_OF_BOUNDS_TIMEDELTA_ERRORS:
                 raise ValueError(
                     "The time difference exceeds the range of values "
                     "that can be expressed at the nanosecond resolution."
@@ -579,7 +574,7 @@ class CFTimeIndex(pd.Index):
     def __rsub__(self, other):
         try:
             return pd.TimedeltaIndex(other - np.array(self))
-        except OUT_OF_BOUNDS_TIMEDELTA_ERROR:
+        except OUT_OF_BOUNDS_TIMEDELTA_ERRORS:
             raise ValueError(
                 "The time difference exceeds the range of values "
                 "that can be expressed at the nanosecond resolution."

--- a/xarray/coding/cftimeindex.py
+++ b/xarray/coding/cftimeindex.py
@@ -43,6 +43,7 @@ import re
 import warnings
 from datetime import timedelta
 from distutils.version import LooseVersion
+from typing import Tuple, Type
 
 import numpy as np
 import pandas as pd
@@ -59,6 +60,7 @@ ITEMS_IN_REPR_MAX_ELSE_ELLIPSIS = 100
 REPR_ELLIPSIS_SHOW_ITEMS_FRONT_END = 10
 
 
+OUT_OF_BOUNDS_TIMEDELTA_ERRORS: Tuple[Type[Exception], ...]
 if LooseVersion(pd.__version__) > LooseVersion("1.1.0"):
     OUT_OF_BOUNDS_TIMEDELTA_ERRORS = (pd.errors.OutOfBoundsTimedelta, OverflowError)
 else:

--- a/xarray/coding/cftimeindex.py
+++ b/xarray/coding/cftimeindex.py
@@ -61,9 +61,9 @@ REPR_ELLIPSIS_SHOW_ITEMS_FRONT_END = 10
 
 
 OUT_OF_BOUNDS_TIMEDELTA_ERRORS: Tuple[Type[Exception], ...]
-if LooseVersion(pd.__version__) > LooseVersion("1.1.0"):
+try:
     OUT_OF_BOUNDS_TIMEDELTA_ERRORS = (pd.errors.OutOfBoundsTimedelta, OverflowError)
-else:
+except AttributeError:
     OUT_OF_BOUNDS_TIMEDELTA_ERRORS = (OverflowError,)
 
 

